src/layout/Border.js

```diff
@@ -339,28 +369,49 @@
         return retVal;
     };
 
+    Border.prototype.checkHandles = function () {
+        var context = this;
+        this.element().selectAll("div.borderHandle")
+            .each(function(handle){
+                var h = d3.select(this);
+                var cell = context.contentDiv.selectAll("#" + context.id() + " > div > div.border-cell-"+handle);
+                h.classed("borderHandleDisabled",cell.node() === null);
+            });
+    };
+    
     Border.prototype.enter = function (domNode, element) {
         HTMLWidget.prototype.enter.apply(this, arguments);
         element.style("position", "relative");
-        this.contentDiv = element.append("div");
+        this.contentDiv = element.append("div").classed("border-content",true);
         this._scrollBarWidth = this.getScrollbarWidth();
+        this._borderHandles = ["top","left","right","bottom"];
+        
+        var handles = element.selectAll("div.borderHandle").data(this._borderHandles);
+        handles.enter().append("div")
+            .classed("borderHandle",true)
+            .each(function(handle){
+                var h = d3.select(this);
+                h.classed("borderHandle_"+handle,true);
+            });
     };
-
+    
     Border.prototype.update = function (domNode, element) {
         HTMLWidget.prototype.update.apply(this, arguments);
         this._sectionTypeArr = this.sectionTypes();
         var context = this;
 
+        element.classed("design-mode",this.designMode());
+
         this.setLayoutOffsets();
 
         var rows = this.contentDiv.selectAll(".cell_" + this._id).data(this.content(), function (d) { return d._id; });
         rows.enter().append("div")
-            .attr("class", "cell_" + this._id)
+            .classed("cell_" + this._id,true)
             .style("position", "absolute")
-            .each(function (d) {
-                d
-                   .target(this)
-                ;
+            .each(function (d,i) {
+                d3.select(this).classed("border-cell border-cell-"+context._sectionTypeArr[i],true);
+                d.target(this);
+                context.checkHandles();
```

Comment:
It's only on enter of a new border cell so it's minimally invasive, but you're right... it would be better to disable the handles within *clearContent*